mkdocs/themes/mkdocs/base.html

```diff
@@ -65,16 +69,16 @@
     </head>
 
     <body{% if page and page.is_homepage %} class="homepage"{% endif %}>
-        <div class="navbar fixed-top navbar-expand-lg navbar-{% if config.theme.nav_style == "light" %}light{% else %}dark{% endif %} bg-{{ config.theme.nav_style }}">
-            <div class="container">
+        <div class="navbar fixed-top navbar-expand-lg bg-{{ config.theme.nav_style }}" data-bs-theme="{% if config.theme.nav_style == 'light' %}light{% else %}dark{% endif %}">
```

Comment:
Hm this comment somehow got lost from the previous batch.

I see this `{% if config.theme.nav_style` in three places and I finally get that it tries to achieve inverted colors for the navbar. But these will not work, they aren't being updated dynamically.
Instead let's keep using the deprecated "navbar-dark" class. (That deprecation seems misguided for sure...)

```suggestion
        <div class="navbar fixed-top navbar-expand-lg navbar-{% if config.theme.nav_style == "light" %}light{% else %}dark{% endif %} bg-{{ config.theme.nav_style }}">
```

Reply:
> But these will not work, they aren't being updated dynamically.

I know they aren't updated dynamically. `nav_style` is not affected by `color_mode` at all and I am not making any reference to color mode here. So there is nothing to be updated dynamically. My suggested change implements the existing behavior with Bootstraps new API as near as we can.

You seem to be suggesting we revert to Bootstraps now-deprecated old API. Why?

Reply:
Hmm something was causing problems for me but now I'm not fully sure that it was this. But sorry, I probably won't have the energy to come back to this and try to understand why this complex logic always works correctly, when the simpler deprecated setup works in a more straightforward way. *And* is also what the preexisting code does - basically I'm reducing the diff. We can reconsider this edit in the future and double-check everything again from a known working state.

Applied this suggestion in https://github.com/mkdocs/mkdocs/commit/351b87939ddc61bc767e2b869e2a2673fe9af5ae